adapters/decenterads/decenterads.go

```diff
@@ -0,0 +1,132 @@
+package decenterads
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+func (a *adapter) MakeRequests(request *openrtb.BidRequest, _ *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	impressions := request.Imp
+	result := make([]*adapters.RequestData, 0, len(impressions))
+	errs := make([]error, 0, len(impressions))
+
+	for _, impression := range impressions {
+		var bidderExt adapters.ExtImpBidder
+		err := json.Unmarshal(impression.Ext, &bidderExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if len(bidderExt.Bidder) == 0 {
+			errs = append(errs, errors.New("bidder parameters required"))
+			continue
+		}
+		var decenteradsExt openrtb_ext.ExtImpDecenterAds
+		err = json.Unmarshal(bidderExt.Bidder, &decenteradsExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		impExtJSON, err := json.Marshal(decenteradsExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		impression.Ext = impExtJSON
+		request.Imp = []openrtb.Imp{impression}
+		body, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		result = append(result, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpoint,
+			Body:    body,
+			Headers: headers,
+		})
```

Comment:
I notice that the `ext` processing is focused on moving `ext.bidder` to `ext`. I'm not sure the re-marshalling of `decenteradsExt` provides you any benefit. Is there something I'm not seeing there?

Otherwise, this simpler code might provide better performance:
```
for _, impression := range impressions {
	var impExt map[string]json.RawMessage
	if err := json.Unmarshal(impression.Ext, &impExt); err != nil {
		errs = append(errs, fmt.Errorf("unable to parse bidder parameers: %s", err))
		continue
	}

	bidderExt, bidderExtExists := impExt["bidder"]
	if !bidderExtExists || len(bidderExt) == 0 {
		errs = append(errs, errors.New("bidder parameters required"))
		continue
	}

	impression.Ext = bidderExt
	request.Imp = []openrtb.Imp{impression}
```

Reply:
+